scripts/publish.ts

```diff
@@ -177,16 +177,28 @@ async function run() {
 
   // If a package has a dependency that has been updated, we need to update the
   // package that depends on it as well.
-  packages.forEach((pkg) => {
+  for (const pkg of packages) {
+    const packageJson = await readPackageJson(
+      path.resolve(rootDir, 'packages', pkg.packageDir, 'package.json'),
+    )
+    const allDependencies = Object.keys(
+      Object.assign(
+        {},
+        packageJson.dependencies ?? {},
+        packageJson.peerDependencies ?? {},
+      ),
+    )
+
     if (
-      pkg.dependencies?.find((dep) =>
```

Comment:
`pkg.dependencies` does not exist - the type were wrong
so we need to find them in packgage.json